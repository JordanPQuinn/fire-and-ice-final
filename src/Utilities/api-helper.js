export const fetchApi = async (url) => {
  const initialFetch = await fetch(url)
  const response = await initialFetch.json();
  return response;
}