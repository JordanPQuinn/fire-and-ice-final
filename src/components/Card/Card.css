.Card {
  border-radius: 10px;
  border-color: black;
  border-style: solid;
  margin: auto;
  padding: 5px;
  max-width: 275px;
  min-height: 550px;
  max-height: 550px;
  overflow-y: auto;
}
