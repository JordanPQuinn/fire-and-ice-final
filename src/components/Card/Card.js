import React, { Component } from 'react';
import './Card.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


export class Card extends Component {
  constructor(props) {
    super(props)

    this.state = {
      clicked: false
    }
  }

  changeClickState = () => {
    this.props.handleClick(this.props.swornMembers);
    const { clicked } = this.state
    this.setState({
      clicked: !clicked
    })
  }

  displayMembers = (members) => {
    return members.map( member => <p key={member.name + member.died}> {member.name}: {member.died} </p> )
  }

  foundedDisplay = (founded) => {
    return !founded ? 
      <h2> Founded: N/A </h2> :
      <h2> Founded: { founded } </h2>
  }

  render() {
    const {name, founded, seats, titles, coatOfArms, ancestralWeapons, words} = this.props;
    const titlesLines = titles.map ( title => {
      return <p key={title}> Titles: {title} </p>
    });

    return(
      <div className='Card' onClick={this.changeClickState}>
        <h1> {name} </h1>
        {this.foundedDisplay(founded)}
        <h2> {words} </h2>
        <p> Seats: {seats} </p>
        <div>
          {titlesLines}
        </div>
        <p> Coat of Arms: {coatOfArms} </p>
        <p> {ancestralWeapons} </p>
        { this.state.clicked && this.displayMembers(this.props.members) }
      </div>
    )
  } 
}

Card.propTypes = {
  house: PropTypes.shape({
    name: PropTypes.string,
    seats: PropTypes.array,
    coatOfArms: PropTypes.string,
    titles: PropTypes.array,
    ancestralWeapons: PropTypes.array,
    swornMembers: PropTypes.array
  })
};

export const mapStateToProps = state => ({
  members: state.members
});

export default connect(mapStateToProps, null)(Card);