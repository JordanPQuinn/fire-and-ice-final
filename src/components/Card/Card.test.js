import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';
import * as api from '../../Utilities/api-helper';
import { mockData } from '../../Utilities/mockData'

describe('Card Component', () => {
  it('should exist', () => {
    const wrapper = shallow(<Card titles={mockData.houses[0].titles} />)
    
    expect(wrapper).toMatchSnapshot();
  });
  it('should change state on a click', () => {
    api.getSwornMembers = () => (mockData.houses[0].swornMembers)
    const wrapper = shallow(
      <Card 
        titles={mockData.houses[0].titles} 
        members={mockData.houses[0].swornMembers}
        storeSwornMembers={jest.fn()} 
      />)

    wrapper.simulate('click');
    wrapper.update();
    expect(wrapper.state().clicked).toEqual(true);
  })

})