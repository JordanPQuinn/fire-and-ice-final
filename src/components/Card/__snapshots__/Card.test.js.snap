// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    titles={
        Array [
            "Lord of Heart's Home",
            "King of the Fingers (historical)",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           
           
</h1>,
        <h2>
           Founded: N/A 
</h2>,
        <h2>
           
           
</h2>,
        <p>
           Seats: 
           
</p>,
        <div>
          <p>
                     Titles: 
                    Lord of Heart's Home
                     
          </p>
          <p>
                     Titles: 
                    King of the Fingers (historical)
                     
          </p>
</div>,
        <p>
           Coat of Arms: 
           
</p>,
        <p>
           
           
</p>,
        false,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Founded: N/A ",
        },
        "ref": null,
        "rendered": " Founded: N/A ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Seats: ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Seats: ",
          undefined,
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
               Titles: 
              Lord of Heart's Home
               
</p>,
            <p>
               Titles: 
              King of the Fingers (historical)
               
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lord of Heart's Home",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Titles: ",
                "Lord of Heart's Home",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " Titles: ",
              "Lord of Heart's Home",
              " ",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "King of the Fingers (historical)",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Titles: ",
                "King of the Fingers (historical)",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " Titles: ",
              "King of the Fingers (historical)",
              " ",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Coat of Arms: ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Coat of Arms: ",
          undefined,
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "p",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             
             
</h1>,
          <h2>
             Founded: N/A 
</h2>,
          <h2>
             
             
</h2>,
          <p>
             Seats: 
             
</p>,
          <div>
            <p>
                         Titles: 
                        Lord of Heart's Home
                         
            </p>
            <p>
                         Titles: 
                        King of the Fingers (historical)
                         
            </p>
</div>,
          <p>
             Coat of Arms: 
             
</p>,
          <p>
             
             
</p>,
          false,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Founded: N/A ",
          },
          "ref": null,
          "rendered": " Founded: N/A ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Seats: ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Seats: ",
            undefined,
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                 Titles: 
                Lord of Heart's Home
                 
</p>,
              <p>
                 Titles: 
                King of the Fingers (historical)
                 
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lord of Heart's Home",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Titles: ",
                  "Lord of Heart's Home",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " Titles: ",
                "Lord of Heart's Home",
                " ",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "King of the Fingers (historical)",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Titles: ",
                  "King of the Fingers (historical)",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " Titles: ",
                "King of the Fingers (historical)",
                " ",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Coat of Arms: ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Coat of Arms: ",
            undefined,
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "p",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
